core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1338,7 +1338,8 @@ class ReplicaManager(val config: KafkaConfig,
         return Some(createLogReadResult(e))
     }
 
-    val remoteFetch = new DelayedRemoteFetch(remoteFetchTask, remoteFetchResult, remoteFetchInfo,
+    val timeout = config.requestTimeoutMs.toLong
+    val remoteFetch = new DelayedRemoteFetch(remoteFetchTask, remoteFetchResult, remoteFetchInfo, timeout,
```

Comment:
Passing the full request timeout to the `DelayedRemoteFetch`, we can reduce the time spent by the handler from the `requestTimeout` to reach up-to here to expire the delayedRemoteFetch requests immediately if required.